tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/BRunUtil.java

```diff
@@ -286,14 +289,25 @@ protected static void invokePackageInit(CompileResult compileResult, String pack
 
         try {
             Method method = clazz.getDeclaredMethod(functionName, jvmParamTypes);
-            jvmResult = method.invoke(null, jvmArgs);
-        } catch (InvocationTargetException e) {
-            Throwable t = e.getTargetException();
-            if (t instanceof BLangRuntimeException) {
-                throw (BLangRuntimeException) t;
-            }
-            throw new RuntimeException("Error while invoking function '" + functionName + "'", e);
-        } catch (NoSuchMethodException | IllegalAccessException e) {
+            Function<Object[], Object> func = a -> {
+                try {
+                    return method.invoke(null, a);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Error while invoking function '" + functionName + "'", e);
+                } catch (InvocationTargetException e) {
+                    Throwable t = e.getTargetException();
+                    if (t instanceof BLangRuntimeException) {
+                        throw (BLangRuntimeException) t;
+                    }
+                    throw new RuntimeException("Error while invoking function '" + functionName + "'", e);
+                }
+            };
+
+            Scheduler scheduler = Scheduler.getInstance();
+            FutureValue futureValue = scheduler.schedule(jvmArgs, func);
+            scheduler.execute();
+            jvmResult = futureValue.result;
```

Comment:
These steps are needed to get the value back from invoking the main method? Do we do the scheduling here? Or the scheduling happens with generated code?

Reply:
These are needed to get back the value. We do the ``scheduler.execute();`` at the Java main. Since we do not have such when running tests, have to start the scheduler here.
Scheduling happens in the generated code. Here, ee only schedule the method invoked by the test.